chore: Implement credential type hierarchy (#3738)
mongodb/terraform-provider-mongodbatlas: .changelog/3738.txt

```release-note:bug

> provider: Enforces strict hierarchy when selecting the credential source (AWS Secrets Manager, provider attributes or environment variables) to prevent mix-and-match behavior where some values could be read from different sources

Reviewer: ```suggestion
provider: Enforces strict hierarchy when selecting the credential source such as AWS Secrets Manager, provider attributes, or environment variables to prevent combining with values from different sources
```
